.github/workflows/builds.yml

```
@@ -73,59 +81,93 @@ jobs:
                   qtdeclarative5-dev \
                   qttools5-dev
 
-      - uses: actions/checkout@v3
+      - if: matrix.wayland
+        name: Install libei and libportal pre-reqs
+        run: |
+          apt-get install -y \
+                  ca-certificates \
+                  dbus \
+                  gettext \
+                  git \
+                  libgirepository1.0-dev \
+                  libglib2.0 \
+                  libgtk-3-dev \
+                  libprotobuf-c-dev \
+                  libsystemd-dev \
+                  meson \
+                  protobuf-c-compiler \
+                  protobuf-compiler \
+                  python3-attr \
+                  python3-dbusmock \
+                  python3-jinja2 \
+                  python3-pip \
+                  python3-pytest \
+                  python3-jinja2 \
+                  valac
+
+      # Code checkout steps
+      - name: Checkout input-leap
+        uses: actions/checkout@v3
         with:
+          path: input-leap
           submodules: recursive
+          set-safe-directory: ${{ github.workspace }}
 
-      - name: Work around git safe directory check
-        run: git config --global --add safe.directory $GITHUB_WORKSPACE
+      - name: Get libei v0.99.1 from freedesktop
+        # Manual checkout of libinput/libei ref 0.99.1 from https://gitlab.freedesktop.org
+        # because actions/checkout does not support gitlab
+        if: matrix.wayland
+        run: |
+          git clone --depth=1 --branch="$ref" --recurse-submodules -- \
+            "https://gitlab.freedesktop.org/libinput/libei" libei
+        env:
+          ref: 0.99.1
 
-      - if: matrix.os == 'ubuntu:22.04'
+      - name: Get libportal from whot/libportal
+        uses: actions/checkout@v3
+        if: matrix.wayland
+        with:
+          repository: whot/libportal
+          ref: wip/inputcapture
+          path: libportal
+
+      - if: matrix.wayland
         name: build libei from git tag (0.99.1)
         run: |
-            apt-get install -y python3-pip
-            pip3 install meson
-            apt-get install -y libsystemd-dev meson git ca-certificates python3-pytest python3-attr python3-dbusmock python3-structlog python3-jinja2
-            git clone --depth 1 --branch 0.99.1 https://gitlab.freedesktop.org/libinput/libei
-            cd libei
-            meson -Dprefix=/usr -Dtests=disabled -Dliboeffis=disabled -Ddocumentation=[] _libei_builddir
+            meson -Dprefix=/usr -Dtests=disabled -Dliboeffis=disabled -Ddocumentation=[] libei _libei_builddir
```

```suggestion
            meson setup -Dprefix=/usr -Dtests=disabled -Dliboeffis=disabled -Ddocumentation=[] libei _libei_builddir
```
using just `meson` is now deprecated, it's always `meson setup` now

---

Meson 1.0.1 indeed prints
```
WARNING: Running the setup command as `meson [options]` instead of `meson setup [options]` is ambiguous and deprecated.
```